### PR TITLE
feat(select): creatable

### PR DESCRIPTION
**Pull request by tajo**
The **creatable select** enables users to create new options along with choosing existing options. This can be also combined with the multi option.

creatable

### CODE REVIEW

File: src/select/select.js [hunk 1]
```diff
@@ -402,16 +402,24 @@ class Select extends React.Component<PropsT, SelectStateT> {
...
+        $selected: boolean,
+        $disabled: boolean,
+        $isHighlighted: boolean,
+      },
     },
-  }): React.Node => {
+  ): React.Node => {
+    if (option.isCreatable) {
+      return (
+        locale.select.create + ' ' + '"' + option[this.props.labelKey] + '"'
```
Comment (gergelyke) [inline src/select/select.js]: would a template literal be more readable here?

Comment (tajo) [reply src/select/select.js]: Yea but it somehow kills code highlighting in my VS Code 😞

**Comment by chasestarr** [comment]
the code LGTM overall - do you mind adding a multi-pick example? That's the example shown in `react-select` here: https://jedwatson.github.io/react-select/

**Comment by tajo** [comment]
> https://jedwatson.github.io/react-select/

**Comment by tajo** [comment]
> the code LGTM overall - do you mind adding a multi-pick example? That's the example shown in `react-select` here: https://jedwatson.github.io/react-select/

Added

File: src/select/select.js [hunk 2]
```diff
@@ -736,20 +744,39 @@ class Select extends React.Component<PropsT, SelectStateT> {
...
+        opt =>
+          opt[this.props.labelKey].toLowerCase() !==
+          filterValue.toLowerCase().trim(),
+      )
+    ) {
+      // $FlowFixMe
+      options.push({
+        id: filterValue,
+        [this.props.labelKey]: filterValue,
+        [this.props.valueKey]: filterValue,
```
Comment (nadiia) [inline src/select/select.js]: `valueKey` is the `id` by default

**Comment by nadiia** [comment]
Similar issue when typed in same as already selected value 
creatable2

File: src/select/select.js (hunk 2 above)
Comment (tajo) [reply src/select/select.js]: Hm, if `valueKey = id` then it just gets overridden and everything is fine?

Or we just don't need `id` and should delete the

```js
id: filterValue
```

row?

Comment (nadiia) [reply src/select/select.js]: Yes, it still works as expected, just duplicated keys 🤷‍♀️

**Pull request merged**